catalog/core/catalog-core-backupplugin/src/main/java/ddf/catalog/backup/CatalogBackupPlugin.java

```diff
@@ -59,55 +60,72 @@
 
     private int subDirLevels;
 
+    private PeriodicBatchExecutor<Metacard> periodicDrain;
+
     private boolean enableBackupPlugin = true;
 
     public CatalogBackupPlugin() {
+
+        init();
+    }
+
+    private void init() {
         subDirLevels = 0;
+        setPeriodicDrain(new PeriodicBatchExecutor<>(1000, 60, TimeUnit.SECONDS));
+        getPeriodicDrain().execute(this::backup);
     }
 
     /**
      * Backs up created metacards to the file system backup.
      *
      * @param input the {@link CreateResponse} to process
-     * @return
+     * @return {@link CreateResponse}
      * @throws PluginExecutionException
      */
     @Override
-    public CreateResponse process(CreateResponse input) throws PluginExecutionException {
+    public synchronized CreateResponse process(CreateResponse input)
```

Comment:
My only question here would be that if we don't take updates and deletes into consideration, how we can assume we have a complete and accurate backup? Let's see what @stustison has to say.